universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -27,29 +32,45 @@
  * normally you need to be logged out in order to be redirected to a login page.
  */
 public class LoginController extends Controller {
+
   private final OidcClient idcsClient;
 
   private final OidcClient adClient;
 
+  private final SAML2Client loginRadiusClient;
+
   private final SessionStore sessionStore;
 
   private final HttpActionAdapter httpActionAdapter;
 
   private final Config config;
 
+  private final String applicantIdp;
+
   @Inject
   public LoginController(
       @AdOidcClient @Nullable OidcClient adClient,
       @IdcsOidcClient @Nullable OidcClient idcsClient,
+      @LoginRadiusSamlClient @Nullable SAML2Client loginRadiusClient,
       SessionStore sessionStore,
       Config config) {
     this.idcsClient = idcsClient;
     this.adClient = adClient;
+    this.loginRadiusClient = loginRadiusClient;
     this.sessionStore = Preconditions.checkNotNull(sessionStore);
+    this.applicantIdp = checkNotNull(config).getString("auth.applicant_idp");
     this.httpActionAdapter = PlayHttpActionAdapter.INSTANCE;
     this.config = config;
   }
 
+  public Result loginWithRedirect(Http.Request request, Optional<String> redirectTo) {
+    if (applicantIdp.equals(AuthIdentityProviderName.LOGIN_RADIUS_APPLICANT.getString())) {
+      return loginRadiusLoginWithRedirect(request, redirectTo);
+    } else {
```

Comment:
can unwrap the else since it's terminal.

Reply:
fixed